File: src/dataset/typescript/allergies_exercism-wackerow/gemini/GeminiFlash_allergies_v4.ts
export default class Allergies {
  private _allergies: number;
  private static readonly ALLERGENS: string[] = [
    'eggs',
    'peanuts',
    'shellfish',
    'strawberries',
    'tomatoes',
    'chocolate',
    'pollen',
    'cats',
  ];

  constructor(score: number) {
    this._allergies = score % 256;
  }

  public allergicTo(allergen: string): boolean {
    const index = Allergies.ALLERGENS.indexOf(allergen);
    if (index === -1) {
      return false;
    }
    return (this._allergies & (1 << index)) !== 0;
  }

  public list(): string[] {
    const result: string[] = [];
    for (let i = 0; i < Allergies.ALLERGENS.length; i++) {
      if ((this._allergies & (1 << i)) !== 0) {
        result.push(Allergies.ALLERGENS[i]);
      }
    }
    return result;
  }
}

File: src/dataset/cpp/allergies_exercism-tqa236/gemini/GeminiFlash_allergies_v4.cpp
#include "allergies.h"

#include <algorithm>
#include <bitset>
#include <map>
#include <unordered_set>

namespace allergies {

const std::map<int, std::string> allergen_value = {
    {1, "eggs"},      {2, "peanuts"},    {4, "shellfish"}, {8, "strawberries"},
    {16, "tomatoes"}, {32, "chocolate"}, {64, "pollen"},   {128, "cats"}};

allergy_test::allergy_test(int score) : score(score) {}

bool allergy_test::is_allergic_to(const std::string& allergen) const {
  return allergies.count(allergen) > 0;
}

std::unordered_set<std::string> allergy_test::get_allergies() {
  allergies.clear();
  int remaining_score = score;
  for (const auto& pair : allergen_value) {
    if (remaining_score >= pair.first) {
      if ((score & pair.first) != 0) {
        allergies.insert(pair.second);
      }
    }
  }
  return allergies;
}

}  // namespace allergies